h2/connection.py

```diff
@@ -1634,6 +1634,33 @@ def _receive_push_promise_frame(self, frame):
 
         return frames, events + stream_events
 
+    def _handle_data_on_closed_stream(self, events, exc, frame):
+        # This stream is already closed - and yet we received a DATA frame.
+        # The received DATA frame counts towards the connection flow window.
+        # We need to manually to acknowledge the DATA frame to update the flow
+        # window of the connection. Otherwise the whole connection stalls due
+        # the inbound flow window being 0.
+        if self._stream_is_closed_by_reset(exc.stream_id):
+            frames = []
+            if frame.flow_controlled_length > 0:
+                self._inbound_flow_control_window_manager.process_bytes(
```

Comment:
`process_bytes()` method works in an interesting (optimised) way, it returns window increment – this value should be used to send `WINDOW_UPDATE` frame. You can take a look into `H2Connection.acknowledge_received_data()` code for a reference.